packages/react/src/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,33 +1,356 @@
+import Details from '../Details'
+import {ActionList} from '../ActionList'
+import {IconButton} from '../Button/IconButton'
+import {KebabHorizontalIcon} from '@primer/octicons-react'
+import {useResizeObserver} from '../hooks/useResizeObserver'
+import type {ResizeObserverEntry} from '../hooks/useResizeObserver'
+import {useOnEscapePress} from '../hooks/useOnEscapePress'
+import {useOnOutsideClick} from '../hooks/useOnOutsideClick'
+import {useFeatureFlag} from '../FeatureFlags'
 
 const SELECTED_CLASS = 'selected'
 
 export type BreadcrumbsProps = React.PropsWithChildren<
   {
     className?: string
+    overflow?: 'wrap' | 'menu' | 'menu-with-root'
   } & SxProp
 >
 
 const BreadcrumbsList = ({children}: React.PropsWithChildren) => {
   return <ol className={classes.BreadcrumbsList}>{children}</ol>
 }
 
-function Breadcrumbs({className, children, sx: sxProp}: BreadcrumbsProps) {
+type BreadcrumbsMenuItemProps = {
+  items: React.ReactElement[]
+  'aria-label'?: string
+}
+
+const BreadcrumbsMenuItem = React.forwardRef<HTMLDetailsElement, BreadcrumbsMenuItemProps>(
+  ({items, 'aria-label': ariaLabel, ...rest}, menuRefCallback) => {
+    const [isOpen, setIsOpen] = useState(false)
+    const detailsRef = useRef<HTMLDetailsElement | null>(null)
+    const detailsRefCallback = useCallback(
+      (element: HTMLDetailsElement | null) => {
+        detailsRef.current = element
+        if (typeof menuRefCallback === 'function') {
+          menuRefCallback(element)
+        }
+      },
+      [menuRefCallback],
+    )
+    const handleSummaryClick = useCallback(
+      (event: React.MouseEvent) => {
+        // Prevent the button click from bubbling up and interfering with details toggle
+        event.preventDefault()
+        // Manually toggle the details element
+        if (detailsRef.current) {
+          const newOpenState = !detailsRef.current.open
+          detailsRef.current.open = newOpenState
+          setIsOpen(newOpenState)
+        }
+      },
+      [detailsRef],
+    )
+
+    const closeOverlay = useCallback(() => {
+      if (detailsRef.current) {
+        detailsRef.current.open = false
+        setIsOpen(false)
+      }
+    }, [detailsRef])
+
+    const focusOnMenuButton = useCallback(() => {
+      iconButtonRef.current?.focus()
+    }, [])
+
+    const iconButtonRef = useRef<HTMLButtonElement>(null)
+    const menuContainerRef = useRef<HTMLDivElement>(null)
+
+    useOnEscapePress(
+      (event: KeyboardEvent) => {
+        if (isOpen) {
+          event.preventDefault()
+          closeOverlay()
+          focusOnMenuButton()
+        }
+      },
+      [isOpen],
+    )
+
+    useOnOutsideClick({
+      onClickOutside: closeOverlay,
+      containerRef: menuContainerRef,
+      ignoreClickRefs: [iconButtonRef],
+    })
+
+    return (
+      <Details ref={detailsRefCallback} className={classes.MenuDetails}>
+        <Details.Summary className={classes.MenuSummary} tabIndex={-1}>
```

Comment:
I wonder if you could actually give the `IconButton` an `as="summary"` so that all the button logic just gets used as the summary element? Rather than wrapping the button in a summary.

Reply:
oh! that works thanks! I can remove that `tabIndex` now